opt/recompose

```diff
@@ -64,44 +57,62 @@ machine api.heroku.com
   password $HEROKU_API_KEY
 EOF
 
-echo "" >> Appfile
 
-echo "Reading primary parts Appfile..."
-apps=0
-while read app_release; do
-  if [ -n "$app_release" ] && [ "$app_release" != "---" ]; then
-    process_slug "$app_release"
-    apps=$((apps+1))
-  fi
-done </app/Appfile
-
-echo "--> Done reading primary parts from Appfile"
-wait_for_downloads $apps "All primary downloads finished"
-
-echo "Reading overlay parts from Appfile..."
-overlay=0
-while read app_release; do
-  if [ -n "$app_release" ] && [ $overlay = 1 ]; then
-    process_slug "$app_release"
-  elif [ "$app_release" = "---" ]; then
-    overlay=1
-  fi
-done </app/Appfile
-if [ $overlay = 1 ]; then
-  echo "--> Done reading overlay parts from Appfile"
-  wait_for_downloads 1 "All overlay downloads finished"
-else
-  echo "No overlay parts found"
-fi
+############ Downloading jdk1.8.0_131_x64,  recomposer.jar #######################################
+JAVA_VERSION="$(basename $JAVA_HOME)"
+mkdir -p $JAVA_HOME
+echo "**** Downloading $JAVA_VERSION from s3..."
+${AWS_CMD:?} s3 cp s3://${AWS_TOOL_S3_BUCKET_NAME:?}/$JAVA_VERSION.tar.gz - | tar xzm --strip-components=1 -C $JAVA_HOME 2>&1
+echo "Completed downloading jdk1.8.0_131_x64 from s3Key:d5eab6c07320844df2961afd84f95df8"
```

Comment:
fyi, u131 has some CVEs

Reply:
yeah, would be better if we didn't download that...

Reply:
btw, I accidentally put this in as a PR to the wrong repo... we've had a bunch of forks. But thanks for merging ;)